### PR TITLE
Edit Payment Requests

### PR DESCRIPTION
**Pull request by ghickman**
This adds part of the edit Payment Requests flow.  I held off tackling the removal of existing files until #1441 has been handled.

### CODE REVIEW

File: opentech/apply/activity/templates/messages/email/payment_request_updated.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% extends "messages/email/applicant_base.html" %}
+
+{% block content %}
+An OTF staff member has edited a Payment Request
```
Comment (todd-dembrey) [inline opentech/apply/activity/templates/messages/email/payment_request_updated.html]:
```
An OTF staff member has updated your payment request for {{ payment_request.project }} for period {{ payment_request.date_from }} to {{ payment_request.date_to }}. It is now {{ payment_request.get_status_display }}.
```

or something like that

File: opentech/apply/activity/tests/test_messaging.py [hunk 2]
```diff
@@ -642,3 +650,51 @@ def test_slack_lead_change(self):
...
+        data = json.loads(responses.calls[0].request.body)
+        self.assertIn(str(applicant), data['message'])
+        self.assertIn(str(project), data['message'])
+
+    @override_settings(
+        SLACK_DESTINATION_URL=target_url,
+        SLACK_DESTINATION_ROOM=target_room,
+    )
+    @responses.activate
+    def test_slack_staff_update_payment_request(self):
```
Comment (todd-dembrey) [inline opentech/apply/activity/tests/test_messaging.py]: Staff should be updated via slack when this happens. It was absent from the requirements added now

It is correct that applicants dont get an email when they submit though.

File: opentech/apply/projects/forms.py [hunk 3]
```diff
@@ -107,6 +108,26 @@ def __init__(self, user=None, *args, **kwargs):
...
+        fields = ['invoice', 'value', 'date_from', 'date_to', 'receipt_list', 'comment']
+        model = PaymentRequest
+
+    def __init__(self, user=None, instance=None, *args, **kwargs):
+        super().__init__(*args, instance=instance, **kwargs)
+
+        self.instance = instance
+
+        self.fields['receipt_list'].choices = [
+            (r.pk, os.path.basename(r.file.url))
```
Comment (todd-dembrey) [inline opentech/apply/projects/forms.py]: Should be working with the `file.name` because storage messes with the basename in the url?

Comment (ghickman) [reply opentech/apply/projects/forms.py]: I'm getting the same output for `file.name` and `file.url` after passing through `os.path.basename`.  I looked into `SubmissionStreamFileField` which has a `filename` but that's the same output for me locally too so doesn't gain us anything.

File: opentech/apply/activity/templates/messages/email/payment_request_updated.html (hunk 1 above)
Comment (ghickman) [reply opentech/apply/activity/templates/messages/email/payment_request_updated.html]: Fixed and rebased.

File: opentech/apply/activity/tests/test_messaging.py (hunk 2 above)
Comment (ghickman) [reply opentech/apply/activity/tests/test_messaging.py]: Fixed and rebased.

**Pull request merged**